### ISSUE TITLE
Reproduce PTB results

### ISSUE THREAD

**Issue by zhang-jian** [benkrause/dynamic-evaluation#1]
Hi,

Thanks for sharing this work.

I tried to reproduce the PTB results (the first two bold rows in Table 1):
<table>
  <tr>
    <th>architecture</th>
    <th>val</th> 
    <th>test</th>
  </tr>
  <tr>
    <td>LSTM(ours)</td>
    <td>88.0</td> 
    <td>85.0</td>
  </tr>
  <tr>
    <td>LSTM+dynamic eval</td>
    <td>73.5</td> 
    <td>71.7</td>
  </tr>
</table>

However, I am getting results:
<pre>
python main.py --cuda --tied           			#valid ppl    90.69 and test ppl    87.08
python dynamiceval.py --model model.pt --val        #perplexity of 79.7848
python dynamiceval.py --model model.pt                #perplexity of 78.9188
</pre>
where main.py (the baseline) is from https://github.com/pytorch/examples/tree/master/word_language_model, and 
dynamiceval.py is from this repository.

There is over 13 ppl difference (from 85.0 to 71.7) on the test data reported in the paper. However, my results show that there is 7.3 ppl difference (from 87.08 to 79.7848).

Did I do anything wrong?

Thanks,

**Comment by benkrause** [comment]
So the default hyper parameters in the code are to reproduce the AWD-LSTM results, with a base model trained using this repository https://github.com/salesforce/awd-lstm-lm, so these should hopefully be easier to reproduce. We did the baseline experiments which you are referring to in Chainer, which has a different LM tutorial from pytorch, and the best hyperparameters for dynamic evaluation were also somewhat different from the AWD-LSTM experiments. Furthermore, I think the default settings in the pytorch LM tutorial use a very small model/hidden state size, which may make it difficult to adapt as effectively. 

If you want to get stronger results with the pytorch LM tutorial, I'd recommend increasing the hidden state size to 650, reducing the lambda parameter slightly (--lamb), maybe to 0.01 (since a weaker base model works better with a weaker L2 decay), and possibly tuning the learning rate (--lr) a bit on the validation set.

**Comment by zhang-jian** [comment]
Thanks for your information. 
Anyway, the 7.3 ppl difference in my experiment on PTB is a very decent result. Thanks for the work.

**Comment by benkrause** [comment]
I ran a couple experiments, firstly just training a larger model with

python main.py --cuda --emsize 650 --nhid 650 --dropout 0.5 --epochs 40

gives 81.1 test, which improves to 70.0 using dynamiceval.py and the default settings.

Also, it is possible to get better results with the smaller model used in your experiment by using a larger learning rate. For instance with 

python dynamiceval.py --model model.pt --lr 0.004

I am getting a test perplexity of 76.4 so up to a 10.7 ppl improvement. Probably both of these results could be improved slightly with better hyper parameter tuning.
